optimum/intel/openvino/modeling_open_clip.py

```diff
@@ -0,0 +1,680 @@
+        local_files_only: bool = False,
+    ) -> PretrainedConfig:
+        config_path = None
+        config_dir = os.path.join(config_name_or_path, subfolder)
+
+        all_files, _ = TasksManager.get_model_files(
+            config_name_or_path, subfolder=subfolder, cache_dir=cache_dir, revision=revision, token=token
+        )
+
+        config_name = None
+        if cls.OPEN_CLIP_CONFIG_NAME in all_files:
+            config_name = cls.OPEN_CLIP_CONFIG_NAME
+        elif cls.CONFIG_NAME in all_files:
+            config_name = cls.CONFIG_NAME
+
+        if os.path.isdir(config_dir):
+            if config_name is None:
+                raise OSError(
+                    f"neither {cls.OPEN_CLIP_CONFIG_NAME} nor {cls.CONFIG_NAME} was found in {config_dir} local folder"
+                )
+            config_path = os.path.join(config_dir, config_name)
+        else:
+            if config_name:
+                config_path = hf_hub_download(
+                    repo_id=config_name_or_path,
+                    filename=config_name,
+                    subfolder=subfolder,
+                    token=token,
+                    revision=revision,
+                    cache_dir=cache_dir,
+                    force_download=force_download,
+                    local_files_only=local_files_only,
+                )
+            else:
+                open_clip_config = OpenClipForZeroShotImageClassification.find_config_by_hub_url(config_name_or_path)
+
+        if config_path:
+            open_clip_config = {}
+            with open(config_path, "r", encoding="utf-8") as f:
+                open_clip_config = json.load(f)
+
+        model_cfg = open_clip_config.get("model_cfg", open_clip_config)
+
+        text_config_dict = model_cfg.get("text_cfg", None) or model_cfg.get("text_config", None)
+        vision_config_dict = model_cfg.get("vision_cfg", None) or model_cfg.get("vision_config", None)
+
+        config = CLIPConfig(
+            text_config_dict=text_config_dict,
+            vision_config_dict=vision_config_dict,
+            **open_clip_config,
+        )
+
+        return config
+
+    # function is overloaded to avoid fails when calling the TasksManager.infer_library_from_model in OptimizedModel.from_pretrained()
+    # this should be removed when open_clip library support is added to optimum
+    @classmethod
+    def from_pretrained(
+        cls,
+        model_id: Union[str, Path],
+        export: bool = False,
+        config: Optional["PretrainedConfig"] = None,
+        use_auth_token: Optional[Union[bool, str]] = None,
+        token: Optional[Union[bool, str]] = None,
+        revision: Optional[str] = None,
+        force_download: bool = False,
+        cache_dir: str = HUGGINGFACE_HUB_CACHE,
+        subfolder: str = "",
+        local_files_only: bool = False,
+        task: Optional[str] = None,
+        trust_remote_code: bool = False,
+        **kwargs,
+    ):
+        if use_auth_token is not None:
```

Comment:
also what about  automatic conversion if the model wasn't already converted https://github.com/huggingface/optimum-intel/blob/b760a1ddcb663e94b69fbe231e088028262824ac/optimum/intel/openvino/modeling_base.py#L405

Reply:
added